### PR TITLE
V0.4 Restore Command

### PR DESCRIPTION
**Pull request by ChuaPingChan**
Part of #175

### CODE REVIEW

File: src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+import seedu.multitasky.commons.core.index.Index;
+import seedu.multitasky.commons.exceptions.IllegalValueException;
+import seedu.multitasky.logic.commands.RestoreByFindCommand;
+import seedu.multitasky.logic.commands.RestoreByIndexCommand;
+import seedu.multitasky.logic.commands.RestoreCommand;
+import seedu.multitasky.logic.parser.exceptions.ParseException;
+
+// @@author A0126623L reused
+/**
+ * Parses input arguments and creates a new DeleteCommand object
```
Comment (akd13) [inline src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java]: DeleteCommand

File: src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            try {
+                Prefix listIndicatorPrefix = ParserUtil.getMainPrefix(argMultimap, PREFIX_FLOATINGTASK,
+                                                                      PREFIX_DEADLINE, PREFIX_EVENT);
+                Index index = ParserUtil.parseIndex(argMultimap.getValue(listIndicatorPrefix).get());
+                return new RestoreByIndexCommand(index, listIndicatorPrefix);
+            } catch (IllegalValueException ive) {
+                throw new ParseException(ive.getMessage(), ive);
+            }
+
+        } else { // process to delete by find.
```
Comment (akd13) [inline src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java]: delete

Comment (ChuaPingChan) [reply src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java]: ah, noted, thanks.

File: src/test/java/seedu/multitasky/logic/commands/RestoreCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import seedu.multitasky.model.Model;
+import seedu.multitasky.model.ModelManager;
+import seedu.multitasky.model.UserPrefs;
+import seedu.multitasky.model.entry.Entry;
+import seedu.multitasky.model.entry.ReadOnlyEntry;
+import seedu.multitasky.testutil.SampleEntries;
+
+//TODO implement restore by find tests and separate out restore by index / abstract class restore tests from here.
+
+//@@author A0126623L reused
```
Comment (mattheuslee) [inline src/test/java/seedu/multitasky/logic/commands/RestoreCommandTest.java]: for reused code put it as `A0126623L-reused`. it ignores everything after whitespace.

**Review by mattheuslee — APPROVED** [review]

File: src/test/java/seedu/multitasky/logic/commands/RestoreCommandTest.java (hunk 3 above)
Comment (ChuaPingChan) [reply src/test/java/seedu/multitasky/logic/commands/RestoreCommandTest.java]: Noted, thanks.

File: src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import java.util.Set;
+
+import seedu.multitasky.commons.core.index.Index;
+import seedu.multitasky.commons.exceptions.IllegalValueException;
+import seedu.multitasky.logic.commands.RestoreByFindCommand;
+import seedu.multitasky.logic.commands.RestoreByIndexCommand;
+import seedu.multitasky.logic.commands.RestoreCommand;
+import seedu.multitasky.logic.parser.exceptions.ParseException;
+
+// @@author A0126623L reused
```
Comment (mattheuslee) [inline src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java]: missed the tag here

File: src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                                             .replaceAll("\\" + CliSyntax.PREFIX_ESCAPE, "");
+            final String[] keywords = searchString.split("\\s+");
+            final Set<String> keywordSet = new HashSet<>(Arrays.asList(keywords));
+
+            return new RestoreByFindCommand(keywordSet);
+        }
+    }
+    // @@author
+
+    // @@author A0126623L reused
```
Comment (mattheuslee) [inline src/main/java/seedu/multitasky/logic/parser/RestoreCommandParser.java]: here and 26 lines later

**Review by akd13 — APPROVED** [review]

**Comment by ChuaPingChan** [comment]
I'll wait for #198 to merge first, make the necessary updates to restore, and then merge this one.

**Review by kevinLamKB — APPROVED** [review]

**Pull request merged**